felix/bpf/proxy/syncer.go

```diff
@@ -1558,3 +1591,23 @@ func K8sSvcWithReapTerminatingUDP() K8sServicePortOption {
 		s.(*servicePort).reapTerminatingUDP = true
 	}
 }
+
+// NewK8sServicePortWithSelector creates a new k8s ServicePort with a node selector
+func NewK8sServicePortWithSelector(clusterIP net.IP, port int, proto v1.Protocol,
```

Comment:
This looks like it is only used in test code, so should probably live in the `_test.go` file